utilities/transactions/transaction_lock_mgr.cc

```diff
@@ -347,6 +348,8 @@ Status TransactionLockMgr::AcquireWithTimeout(
                          &expire_time_hint, &wait_ids);
 
   if (!result.ok() && timeout != 0) {
+    PERF_TIMER_GUARD(key_lock_wait_time);
```

Comment:
I remember you mentioned that you already have this time computed somewhere. Wouldn't it be more efficient to use that instead re-computing it here? I am guessing there must be an overhead with obtaining the current time in PERF_TIMER_GUARD.

Reply:
We do, and it's stored in `start_time`, but I ended up thinking it might cleaner just to reuse `PERF_TIMER_GUARD`, since it takes care of checking perf level for me.

Reply:
If we do not pay the extra cost of recording the time i would say its perf level is not "time" and it is rather at the level of cheaper Normal stats. The perf level was a handy way to disable time measurement overhead, which we would not incur if we use our existing time values.

Reply:
It doesn't look like `start_time` computes the same thing:
(1) it starts measuring even when the lock is acquired on the first attempt, so it'd give you nonzero `key_lock_wait_time` even when `key_lock_wait_count == 0`;
(2) doesn't consider what time lock is grabbed;
(3) uses `GetMicros()`, which isn't monotonically increasing so taking the difference of two such times could be dangerous